lib/Sema/CSSimplify.cpp

```diff
@@ -1570,14 +1570,17 @@ static bool matchFunctionRepresentations(FunctionTypeRepresentation rep1,
     return rep1 == rep2;
       
   case ConstraintKind::Subtype: {
-    auto last = locator.last();
-    if (!(last && last->is<LocatorPathElt::FunctionArgument>()))
-      return true;
-
+    // Thick(swift or block) to Thin(c or thin) subtype relation e.g.
+    // '(@convention(thin) () -> Void) -> Void' sub '(@convention(block) () ->
+    // Void) -> Void', which in contravariant function argument means a block
+    // sub thin in this case is never allowed, so we can just diagnose this
+    // early on type checking. This is different from conversions that can just
+    // be allowed at type-checking and conversion can be deferred to
+    // CSApply/SILGen where it may be possible on a context basis or diagnosed
+    // appropriately.
```

Comment:
IMO this comment is kinda' re-explaining what is already explained by the doc comment for `isSubtypeOf` and the `[NOTE: diagnose-swift-to-c-convention-change]`. For example, the fact that `@convention(block)` is not a subtype of `@convention(thin)` is made clear by the comment for `isSubtypeOf`.

So I think we should remove this comment, and potentially update the two existing ones that I pointed out if things aren't clear. Having the same information in multiple places makes it easier for it to get out-dated.

Also, you can remove the function parameter `locator` as it is unused.

Reply:
> So I think we should remove this comment, and potentially update the two existing ones that I pointed out if things aren't clear. Having the same information in multiple places makes it easier for it to get out-dated.
> 

Yeah, you are right this is indeed basically the same information. 

> Also, you can remove the function parameter `locator` as it is unused.

True will fix :)

Reply:
Fixed!